components/services/bat_ledger/bat_ledger_impl.cc

```diff
@@ -594,6 +594,17 @@ void BatLedgerImpl::UpdateMediaDuration(
   ledger_->UpdateMediaDuration(window_id, publisher_key, duration, first_visit);
 }
 
+void BatLedgerImpl::IsPublisherRegistered(
+    const std::string& publisher_id,
+    IsPublisherRegisteredCallback callback) {
+  auto shared_callback =
+      std::make_shared<decltype(callback)>(std::move(callback));
+  ledger_->IsPublisherRegistered(
+      publisher_id, [shared_callback](bool is_registered) {
+        std::move(*shared_callback).Run(is_registered);
+      });
+}
```

Comment:
You can create this `std::shared_ptr<>` as a capture, so that we don't make an unnecessary copy of it (hence not increase/decrease the atomic reference counter).
```
void BatLedgerImpl::IsPublisherRegistered(
    const std::string& publisher_id,
    IsPublisherRegisteredCallback callback) {
  ledger_->IsPublisherRegistered(
      publisher_id, [shared_callback = std::make_shared<decltype(callback)>(
                         std::move(callback))](bool is_registered) {
        std::move(*shared_callback).Run(is_registered);
      });
}
```